packages/core/src/domain/report/reportObservable.ts

```diff
@@ -0,0 +1,119 @@
+import { mergeObservables, Observable } from '../../tools/observable'
+import { DOM_EVENT, includes, addEventListener } from '../../tools/utils'
+import { monitor } from '../internalMonitoring'
+import type {
+  Report,
+  BrowserWindow,
+  ReportType,
+  ReportingObserverCallback,
+  ReportingObserverOption,
+  ReportingObserver as ReportingObserverInterface,
+} from './browser.types'
+
+declare const ReportingObserver: {
+  new (callback: ReportingObserverCallback, option: ReportingObserverOption): ReportingObserverInterface
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** could you:

- move the constructor type in the existing ReportingObserver interface
- ensure to use ReportingObservable variable from the BrowserWindow type
- remove the declare const

Reply:
If I declare the ReportingObserver in browser.types like so:
```typescript
export interface BrowserWindow {
  ReportingObserver: ReportingObserver
}

declare const ReportingObserver: {
  prototype: PerformanceEntry
  new (callback: ReportingObserverCallback, option: ReportingObserverOption): ReportingObserver
}

interface ReportingObserver {
  disconnect(): void
  observe(): void
  takeRecords(): Report[]
}
```

TS complains when I try use ReportingObservable variable from the BrowserWindow type:
```typescript
const test = new (window as BrowserWindow).ReportingObserver()
```
TS error message:
`This expression is not constructable. Type 'ReportingObserver' has no construct signatures.`

Reply:
Same if I declare a class like so:
```typescript
declare class ReportingObserver { 
  constructor(callback: ReportingObserverCallback, option: ReportingObserverOption) 
  disconnect(): void 
  observe(): void 
  takeRecords(): Report[] 
}
```

Reply:
what I had in mind: https://github.com/DataDog/browser-sdk/commit/4c522570562a01b4271facaf84873338da6084d0

Reply:
Way nicer indeed!